opm/grid/LookUpData.hh

```diff
@@ -431,6 +464,49 @@ Opm::LookUpData<Grid,GridView>::getFieldPropIdx(const int& elemIdx) const
     }
 }
 
+template<typename Grid, typename GridView>
+template<typename GridType>
+typename std::enable_if_t<!std::is_same_v<GridType,Dune::CpGrid>,int>
+Opm::LookUpData<Grid,GridView>::getParentFaceIdxFromLgrBoundaryFace(const int& faceIdx) const
+{
+    static_assert(std::is_same_v<Grid,GridType>);
+    // Check there are no LGRs. LGRs (level>0) only supported for CpGrid.
+    assert(gridView_.grid().maxLevel() == 0);
+    return faceIdx;
+}
+
+
+template<typename Grid, typename GridView>
+template<typename GridType>
+typename std::enable_if_t<std::is_same_v<GridType,Dune::CpGrid>,int>
+Opm::LookUpData<Grid,GridView>::getParentFaceIdxFromLgrBoundaryFace(const int& faceIdx) const
+{
+    static_assert(std::is_same_v<Grid,GridType>);
+    return gridView_.grid().getParentFaceFromLgrBoundaryFace(faceIdx).index();
+}
+
+template<typename Grid, typename GridView>
+template<typename GridType>
+typename std::enable_if_t<!std::is_same_v<GridType,Dune::CpGrid>,bool>
+Opm::LookUpData<Grid,GridView>::isOnLgrBoundary(const int& faceIdx [[maybe_unused]]) const
+{
+    static_assert(std::is_same_v<Grid,GridType>);
+    // Check there are no LGRs. LGRs (level>0) only supported for CpGrid.
+    assert(gridView_.grid().maxLevel() == 0);
+    return gridView_.grid().maxLevel()>0;
```

Comment:
If we merge this we need a working implementation. This will just run into assert for any refined grid other than CpGrid.
Wouldn't it be possible to use the intersection instead of an index? Then we can simply check the `!intersection.neighbor() || intersection.inside().level() != intersection.outside.level()`
Would work for any DUNE grid.

Reply:
That could work. I didn't notice. In principle, both levels could be different and not zero, but with the assumption of disjoint LGRs, then at least one of them should be zero

Reply:
On the other hand, we could just return false, and remove the unnecessary assert